### PR TITLE
Update Germany holidays: add `th` localization, deprecated non-standard ISO code, add subdivision aliases

### PR DESCRIPTION
**Pull request by PPsyrius**
## Proposed change

This PR includes 3 changes for Germany 🇩🇪  holidays:
- Deprecate the unofficial `BYP` code. Catholic municipalities in Bavaria will instead need to include the `CATHOLIC` category to mimic the previously implemented behavior.
  - Maybe we can do something similar to #1952 by adding "`Bayern (Katholisch)`" as a subregion too? I'm open to all suggestions here.
- Add Subdivision Aliases for all Germany subdivisions.
- Add `th` localization. (Checked with https://bangkok.diplo.de/th-th/botschaft/oeffnungszeiten/1670002 )

## Type of change

- [ ] New country/market holidays support (thank you!)
- [x] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/pin/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've run `make pre-commit`, it didn't generate any changes
- [x] I've run `make test`, all tests passed locally

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/dev/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/dev/docs/source

### CODE REVIEW

File: holidays/constants.py [hunk 1]
```diff
@@ -53,3 +53,4 @@
 HEBREW = "hebrew"
 HINDU = "hindu"
 ISLAMIC = "islamic"
+PROTESTANT = "protestant"
```
Comment (KJhellico) [inline holidays/constants.py:56]: It would be more logical to add the category CATHOLIC. Then Assumption Day will belong to `by_catholic_holidays`.

Comment (PPsyrius) [reply holidays/constants.py:56]: Hmm, true - my only concern is that it changes the default behavior for `BY` subdivision.

Comment (PPsyrius) [reply holidays/constants.py]: Implemented.

**Comment by PPsyrius** [comment]
Hmm, seems like we might be able to consolidate this a bit further:
image

**Review by KJhellico — APPROVED** [review]
LGTM.

**Review by arkid15r — COMMENTED** [review]
🇩🇪 LGTM

**Review by arkid15r — APPROVED** [review]

**Pull request merged**